src/connector/src/sink/deltalake.rs

```diff
@@ -0,0 +1,521 @@
+            Err(SinkError::DeltaLake(
+                "path need to start with 's3://','s3a://'(s3) or file://(local)".to_string(),
+            ))
+        }
+    }
+}
+
+enum DeltaTableUrl {
+    S3(String),
+    Local(String),
+}
+
+#[serde_as]
+#[derive(Clone, Debug, Deserialize, WithOptions)]
+pub struct DeltaLakeConfig {
+    #[serde(flatten)]
+    pub common: DeltaLakeCommon,
+
+    pub r#type: String,
+}
+
+impl DeltaLakeConfig {
+    pub fn from_hashmap(properties: HashMap<String, String>) -> Result<Self> {
+        let config = serde_json::from_value::<DeltaLakeConfig>(
+            serde_json::to_value(properties).map_err(|e| SinkError::DeltaLake(e.to_string()))?,
+        )
+        .map_err(|e| SinkError::Config(anyhow!(e)))?;
+        Ok(config)
+    }
+}
+
+#[derive(Debug)]
+pub struct DeltaLakeSink {
+    pub config: DeltaLakeConfig,
+    param: SinkParam,
+}
+
+impl DeltaLakeSink {
+    pub fn new(config: DeltaLakeConfig, param: SinkParam) -> Result<Self> {
+        Ok(Self { config, param })
+    }
+}
+
+fn check_field_type(rw_data_type: &DataType, dl_data_type: &DeltaLakeDataType) -> Result<bool> {
+    let result = match rw_data_type {
+        DataType::Boolean => {
+            matches!(
+                dl_data_type,
+                DeltaLakeDataType::Primitive(PrimitiveType::Boolean)
+            )
+        }
+        DataType::Int16 => {
+            matches!(
+                dl_data_type,
+                DeltaLakeDataType::Primitive(PrimitiveType::Short)
+            )
+        }
+        DataType::Int32 => {
+            matches!(
+                dl_data_type,
+                DeltaLakeDataType::Primitive(PrimitiveType::Integer)
+            )
+        }
+        DataType::Int64 => {
+            matches!(
+                dl_data_type,
+                DeltaLakeDataType::Primitive(PrimitiveType::Long)
+            )
+        }
+        DataType::Float32 => {
+            matches!(
+                dl_data_type,
+                DeltaLakeDataType::Primitive(PrimitiveType::Float)
+            )
+        }
+        DataType::Float64 => {
+            matches!(
+                dl_data_type,
+                DeltaLakeDataType::Primitive(PrimitiveType::Double)
+            )
+        }
+        DataType::Decimal => {
+            matches!(
+                dl_data_type,
+                DeltaLakeDataType::Primitive(PrimitiveType::Decimal(_, _))
+            )
+        }
+        DataType::Date => {
+            matches!(
+                dl_data_type,
+                DeltaLakeDataType::Primitive(PrimitiveType::Date)
+            )
+        }
+        DataType::Varchar => {
+            matches!(
+                dl_data_type,
+                DeltaLakeDataType::Primitive(PrimitiveType::String)
+            )
+        }
+        DataType::Timestamp => {
+            matches!(
+                dl_data_type,
+                DeltaLakeDataType::Primitive(PrimitiveType::Timestamp)
+            )
+        }
```

Comment:
https://github.com/delta-io/delta-rs/blob/8ca8d655cf4eabc86a748f9b831536d2a6ef28a6/crates/deltalake-core/src/kernel/schema.rs#L427-L428
> Microsecond precision timestamp, adjusted to UTC

This corresponds to `timestamptz` in PostgreSQL/RisingWave